sentry-tracing/src/converters.rs

```diff
@@ -163,13 +182,53 @@ where
     // proper grouping and issue metadata generation. tracing_core::Record does not contain sufficient
     // information for this. However, it may contain a serialized error which we can parse to emit
     // an exception record.
-    let (message, visitor) = extract_event_data(event);
+    let (mut message, visitor) = extract_event_data(event);
+    let FieldVisitor {
+        mut exceptions,
+        mut json_values,
+    } = visitor;
+
+    if !exceptions.is_empty() && message.is_some() {
+        #[allow(unused_mut)]
+        let mut thread = Thread::default();
+
+        #[cfg(feature = "backtrace")]
+        if let Some(client) = sentry_core::Hub::current().client() {
+            if client.options().attach_stacktrace {
+                thread = sentry_backtrace::current_thread(true);
```

Comment:
These two changes I'm not happy with. In order to attach the stack trace, I need to depend on `sentry_backtrace` and also grab the client to check its options. 

The backtrace integration could detect this case (attach_stacktrace enabled + top-level exception is synthetic) and write there. Would that be a better alternative? It composes better, but I'm not sure if we're happy with these semantics.

Reply:
I believe its a fair thing to do. However turning on `attach_stacktrace` is a rather hard hammer, and may have high overhead if you just want to log random messages, and not necessarily errors.

As I said in another comment. Another alternative would be to add the `"Rust Tracing Location"` as it is now as the stack trace here. That is definitely better than it being hidden somewhere in a random context.

Reply:
I was thinking about adding the tracing location, too. However, it doesn't have enough information to create even a single stack frame that is eligible for grouping, most notably because of the missing function name. 

We could also pass this as an option to the tracing integration and pass it down to the converter function, so you can control this separately. At least in Relay, we've enabled stack traces for everything (they are just added for error messages anyway, otherwise we're not tracking stuff in Sentry), so the behavior is intended. This is a good consideration for what attach_stacktraces is actually supposed to mean; and I think in this specific case (we have an exception entry) it's seems exactly as intended.